superset-frontend/packages/superset-ui-core/src/query/types/Query.ts

```diff
@@ -32,23 +32,30 @@ import { PostProcessingRule } from './PostProcessing';
 import { JsonObject } from '../../connection';
 import { TimeGranularity } from '../../time-format';
 
-export type QueryObjectFilterClause = {
+export type BaseQueryObjectFilterClause = {
   col: QueryFormColumn;
   grain?: TimeGranularity;
   isExtra?: boolean;
```

Comment:
the grain and isExtra(if the column is a time column and needs grain) are still available to us, please do not change them. .....or we can refactor this part later.

Reply:
With this refactoring I was trying to keep everything the same except allow the specific types of filter clauses (binary, set, unary) to be exported/imported separately, as TypeScript was complaining because `val` was optional on the union `QueryObjectFilterClause` type.  If I did it right `grain` and `isExtra` should still be present on all specific filter clause types and the union `QueryObjectFilterClause` type – does that work?